QuizzApp/Scenes/HomePage/HomePageViewController.swift

```diff
@@ -0,0 +1,289 @@
+                                                       constant: Constants.subjectsTableViewLeftPadding),
+        ])
+    }
+    
+    // MARK: Separator View Constraints
+    func separatorViewConstraints() {
+        NSLayoutConstraint.activate([
+            separatorView.topAnchor.constraint(lessThanOrEqualTo: subjectsTableView.bottomAnchor,
+                                               constant: Constants.separatorViewTopPadding),
+            separatorView.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            separatorView.leadingAnchor.constraint(equalTo: view.leadingAnchor,
+                                                   constant: Constants.separatorViewLeftPadding),
+            separatorView.heightAnchor.constraint(equalToConstant: Constants.separatorViewHeight)
+        ])
+    }
+    
+    // MARK: Log Out Button Constraints
+    func logOutButtonConstraints() {
+        NSLayoutConstraint.activate([
+            logOutButton.topAnchor.constraint(equalTo: separatorView.bottomAnchor,
+                                              constant: Constants.logOutButtonTopPadding),
+            logOutButton.trailingAnchor.constraint(equalTo: view.trailingAnchor,
+                                                   constant: Constants.logOutButtonRightPadding),
+            logOutButton.bottomAnchor.constraint(equalTo: view.bottomAnchor,
+                                                 constant: Constants.logOutButtonBottomPadding)
+        ])
+    }
+    
+    // MARK: Move To Score View Controller
+    @objc func moveToScore() {
+        let viewController = ScoreViewController()
+        navigationController?.pushViewController(viewController, animated: true)
+    }
+    
+    // MARK: Log Out
+    @objc func logOut() {
+        alertView.modalPresentationStyle = .overFullScreen
+        present(alertView, animated: true)
+    }
+}
+
+// MARK: - Table View Functions
+extension HomePageViewController: UITableViewDelegate, UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        subjects.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: SubjectsTableViewCell.identifier) as? SubjectsTableViewCell else {return UITableViewCell()}
+        let subject = subjects[indexPath.row]
+        cell.configCell(with: subject)
+        
+        return cell
+    }
+    
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        UITableView.automaticDimension
+    }
+    
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+        let viewController = TestViewController()
+        navigationController?.pushViewController(viewController, animated: true)
```

Comment:
აქ viewController-ის declaration აღარაა საჭირო, პირდაპირ გამოიყენე
```suggestion
        navigationController?.pushViewController(TestViewController(), animated: true)
```